src/components/data-table/ha-data-table.ts

```diff
@@ -256,73 +257,85 @@ export class HaDataTable extends LitElement {
               `;
             })}
           </div>
-          <div class="mdc-data-table__content scroller">
-            ${scroll({
-              items: this._filteredData,
-              renderItem: (row: DataTableRowData) => html`
-                <div
-                  .rowId="${row[this.id]}"
-                  @click=${this._handleRowClick}
-                  class="mdc-data-table__row ${classMap({
-                    "mdc-data-table__row--selected": this._checkedRows.includes(
-                      String(row[this.id])
-                    ),
-                  })}"
-                  aria-selected=${ifDefined(
-                    this._checkedRows.includes(String(row[this.id]))
-                      ? true
-                      : undefined
-                  )}
-                  .selectable=${row.selectable !== false}
-                >
-                  ${this.selectable
-                    ? html`
-                        <div
-                          class="mdc-data-table__cell mdc-data-table__cell--checkbox"
-                        >
-                          <ha-checkbox
-                            class="mdc-data-table__row-checkbox"
-                            @change=${this._handleRowCheckboxClick}
-                            .disabled=${row.selectable === false}
-                            .checked=${this._checkedRows.includes(
-                              String(row[this.id])
-                            )}
-                          >
-                          </ha-checkbox>
-                        </div>
-                      `
-                    : ""}
-                  ${Object.entries(this.columns).map((columnEntry) => {
-                    const [key, column] = columnEntry;
-                    return html`
+          ${!this._filteredData.length
+            ? html`
+                <div class="mdc-data-table__content">
+                  <div class="mdc-data-table__row">
+                    <div class="mdc-data-table__cell grows center">
+                      ${this.noDataText || "No data"}
```

Comment:
Can we allow for a slot to be rendered if there is no data ? And we just don't render the virtualizer. That way we can do more graphic calls to action. 

Also, if there are filters and some data is hidden, should we mention that with a call to action to clear current filters?

We also could do both in a future PR btw.